src/model/GeneratorWindTurbine.cpp

```diff
@@ -0,0 +1,639 @@
+    }
+
+    double GeneratorWindTurbine_Impl::heightforLocalAverageWindSpeed() const {
+      boost::optional<double> value = getDouble(OS_Generator_WindTurbineFields::HeightforLocalAverageWindSpeed, true);
+      OS_ASSERT(value);
+      return value.get();
+    }
+
+    boost::optional<double> GeneratorWindTurbine_Impl::bladeChordArea() const {
+      return getDouble(OS_Generator_WindTurbineFields::BladeChordArea, true);
+    }
+
+    double GeneratorWindTurbine_Impl::bladeDragCoefficient() const {
+      boost::optional<double> value = getDouble(OS_Generator_WindTurbineFields::BladeDragCoefficient, true);
+      OS_ASSERT(value);
+      return value.get();
+    }
+
+    double GeneratorWindTurbine_Impl::bladeLiftCoefficient() const {
+      boost::optional<double> value = getDouble(OS_Generator_WindTurbineFields::BladeLiftCoefficient, true);
+      OS_ASSERT(value);
+      return value.get();
+    }
+
+    double GeneratorWindTurbine_Impl::powerCoefficientC1() const {
+      boost::optional<double> value = getDouble(OS_Generator_WindTurbineFields::PowerCoefficientC1, true);
+      OS_ASSERT(value);
+      return value.get();
+    }
+
+    double GeneratorWindTurbine_Impl::powerCoefficientC2() const {
+      boost::optional<double> value = getDouble(OS_Generator_WindTurbineFields::PowerCoefficientC2, true);
+      OS_ASSERT(value);
+      return value.get();
+    }
+
+    double GeneratorWindTurbine_Impl::powerCoefficientC3() const {
+      boost::optional<double> value = getDouble(OS_Generator_WindTurbineFields::PowerCoefficientC3, true);
+      OS_ASSERT(value);
+      return value.get();
+    }
+
+    double GeneratorWindTurbine_Impl::powerCoefficientC4() const {
+      boost::optional<double> value = getDouble(OS_Generator_WindTurbineFields::PowerCoefficientC4, true);
+      OS_ASSERT(value);
+      return value.get();
+    }
+
+    double GeneratorWindTurbine_Impl::powerCoefficientC5() const {
+      boost::optional<double> value = getDouble(OS_Generator_WindTurbineFields::PowerCoefficientC5, true);
+      OS_ASSERT(value);
+      return value.get();
+    }
+
+    double GeneratorWindTurbine_Impl::powerCoefficientC6() const {
+      boost::optional<double> value = getDouble(OS_Generator_WindTurbineFields::PowerCoefficientC6, true);
+      OS_ASSERT(value);
+      return value.get();
+    }
+
+    bool GeneratorWindTurbine_Impl::setAvailabilitySchedule(Schedule& schedule) {
+      bool result = setSchedule(OS_Generator_WindTurbineFields::AvailabilityScheduleName, "GeneratorWindTurbine", "Availability", schedule);
+      return result;
+    }
+
+    void GeneratorWindTurbine_Impl::resetAvailabilitySchedule() {
+      bool result = setString(OS_Generator_WindTurbineFields::AvailabilityScheduleName, "");
+      OS_ASSERT(result);
+    }
+
+    bool GeneratorWindTurbine_Impl::setRotorType(std::string rotorType) {
+      bool result = setString(OS_Generator_WindTurbineFields::RotorType, rotorType);
+      return result;
+    }
```

Comment:
(Not sold on it, but should we clear/zero out the fields that become moot?)

Reply:
We'll leave it up to the user to control the blade fields.